server/vulnerabilities/sync.go

```diff
@@ -0,0 +1,262 @@
+	"github.com/fleetdm/fleet/v4/pkg/fleethttp"
+	"github.com/fleetdm/fleet/v4/server/config"
+	"github.com/fleetdm/fleet/v4/server/fleet"
+	"github.com/fleetdm/fleet/v4/server/ptr"
+	"github.com/hashicorp/go-multierror"
+	"go.uber.org/multierr"
+)
+
+func Sync(vulnPath string, config config.FleetConfig, ds fleet.Datastore) error {
+	if config.Vulnerabilities.DisableDataSync {
+		return nil
+	}
+
+	client := fleethttp.NewClient()
+
+	var syncErr error
+
+	if err := DownloadCPEDatabase(vulnPath, client, WithCPEURL(config.Vulnerabilities.CPEDatabaseURL)); err != nil {
+		syncErr = multierror.Append(syncErr, fmt.Errorf("sync CPE database: %w", err))
+	}
+
+	if err := DownloadNVDCVEFeed(vulnPath, ""); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync NVD CVE feed: %w", err))
+	}
+
+	if err := DownloadEPSSFeed(vulnPath, client); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync EPSS CVE feed: %w", err))
+	}
+
+	if err := DownloadCISAKnownExploitsFeed(vulnPath, client); err != nil {
+		syncErr = multierr.Append(syncErr, fmt.Errorf("sync CISA known exploits feed: %w", err))
+	}
+
+	if err := LoadCVEScores(vulnPath, ds); err != nil {
+		syncErr = multierr.Append(syncErr, err)
+	}
+
+	return syncErr
+}
+
+const epssFeedsURL = "https://epss.cyentia.com"
+const epssFilename = "epss_scores-current.csv.gz"
+
+func DownloadEPSSFeed(vulnPath string, client *http.Client) error {
+	urlString := epssFeedsURL + "/" + epssFilename
+	u, err := url.Parse(urlString)
+	if err != nil {
+		return fmt.Errorf("parse url: %w", err)
+	}
+	path := filepath.Join(vulnPath, strings.TrimSuffix(epssFilename, ".gz"))
+
+	err = download.Download(client, u, path)
+	if err != nil {
+		return fmt.Errorf("download %s: %w", u, err)
+	}
+
+	return nil
+}
+
+// EPSSScore represents the EPSS score for a CVE.
+type EPSSScore struct {
+	CVE   string
+	Score float64
+}
+
+func parseEPSSScoresFile(path string) ([]EPSSScore, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+
+	r := csv.NewReader(f)
+	r.Comment = '#'
+	r.FieldsPerRecord = 3
+
+	// skip the header
+	r.Read()
+
+	var epssScores []EPSSScore
+	for {
+		rec, err := r.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if len(rec) != 3 {
+			continue
+		}
+
+		cve := rec[0]
+		score, err := strconv.ParseFloat(rec[1], 64)
+		if err != nil {
+			return nil, fmt.Errorf("parse epss score: %w", err)
+		}
+
+		// ignore percentile
+
+		epssScores = append(epssScores, EPSSScore{
+			CVE:   cve,
+			Score: score,
+		})
+	}
+
+	return epssScores, nil
+}
+
+const cisaKnownExploitsURL = "https://www.cisa.gov/sites/default/files/feeds/known_exploited_vulnerabilities.json"
+const cisaKnownExploitsFilename = "known_exploited_vulnerabilities.json"
+
+// KnownExploitedVulnerabilitiesCatalog represents the CISA Catalog of Known Exploited Vulnerabilities.
+type KnownExploitedVulnerabilitiesCatalog struct {
```

Comment:
Do we want to make this public? Also, if this is only used for deserialization, maybe we might want to use an anon struct inside the function?